pkg/kapp/app/recorded_app.go

```diff
@@ -40,11 +42,33 @@ type RecordedApp struct {
 	logger       logger.Logger
 }
 
+func NewRecordedApp(name, nsName string, creationTimestamp time.Time, coreClient kubernetes.Interface,
+	identifiedResources ctlres.IdentifiedResources, appInDiffNsHintMsgFunc func(string) string, logger logger.Logger, app *corev1.ConfigMap) *RecordedApp {
+
+	recordedApp := &RecordedApp{name, nsName, false, creationTimestamp, coreClient, identifiedResources, appInDiffNsHintMsgFunc,
+		nil, logger.NewPrefixed("RecordedApp")}
+
+	if app == nil {
+		return recordedApp
+	}
+
+	if _, ok := app.Annotations[KappIsConfigmapMigratedAnnotationKey]; ok && recordedApp.isMigrationEnabled() {
+		recordedApp.name = strings.TrimSuffix(recordedApp.name, AppSuffix)
+		recordedApp.isMigrated = true
+	}
+
+	recordedApp.setMeta(*app)
+
+	return recordedApp
+}
+
```

Comment:
Apps looks fine now, but I feel that the Recorded app constructor is a bit complex now (with the configmap and the usage of private methods)

Reply:
As this code[From L55-L60] is specific to list, how about putting it back there and in here, we can just instantiate recordedApp and return it

Reply:
+1 on what Rohit has said to make it separate from constructor to not to make constructor complex.

Reply:
The idea was to move things related to recorded app from apps to recorded app.